### PR TITLE
[GD32_ARM] Fix bugs in drv_adc when enable multiple channel.

### PR DESCRIPTION
**Pull request by Blues-Jiang**
## 拉取/合并请求描述：(PR description)

[
 Test on GD32F303

#### 为什么提交这份PR (why to submit this PR)

ADC driver of GD32_ARM have multiple bugs.
1. ADC should use single operation mode, but current version use discontinuous operation mode.
2. When calibration is used after turning on the ADC, it is necessary to wait for 14 CK_ADC(about 350ns in GD32F303).
3. When enabling multiple ADC channels at once and sequentially reading the different channels, the values read are all the same.

#### 你的解决方案是什么 (what is your solution)
1. Make sure ADC enable single operation mode.
2. Add a 1us dely after ADCON.
3. Clear EOC flag before start ADC convert to prevent other bugs.
4. The setting of the regular channel should be done at each reading, rather than at the point of enabling.

#### 请提供验证的bsp和config (provide the config and bsp) 

- BSP:GD32/ARM

- .config:

- action:

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

**Comment by mysterywolf** [comment]
请签署CLA

**Comment by Blues-Jiang** [comment]
昨天测了GD32F427发现还有问题，F1~F3应该差不多了，还得改一下。

**Comment by mysterywolf** [comment]
可以合并请告知

**Comment by Blues-Jiang** [comment]
看到有格式错误修了，加了下超时机制。
已在GD32F427ZGT6和GD32F303VET6验证最后版本。我需要修改的都清了，可以合并了。

File: bsp/gd32/arm/libraries/gd32_drivers/drv_adc.c [hunk 1]
```diff
@@ -99,6 +99,15 @@ static void gd32_adc_gpio_init(rcu_periph_enum adc_clk, rt_base_t pin)
 #endif
 }
 
+static void gd32_adc_udelay(rt_uint32_t us)
```
Comment (mysterywolf) [inline bsp/gd32/arm/libraries/gd32_drivers/drv_adc.c:102]: 这个可以用rt_hw_delay_us 函数代替嘛？

Comment (mysterywolf) [reply bsp/gd32/arm/libraries/gd32_drivers/drv_adc.c:102]: 不建议再搞一个这种us延时函数 直接使用rt_hw_delay_us 即可

Comment (Blues-Jiang) [reply bsp/gd32/arm/libraries/gd32_drivers/drv_adc.c:102]: ok,当时考虑的是GD32仓库里本身没有实现这个，用的话当然是都可以的。
一般bsp这个函数初始化在哪里更合适些？我都是自己用的时候会自己加。

Comment (mysterywolf) [reply bsp/gd32/arm/libraries/gd32_drivers/drv_adc.c:102]: 可以参见STM32的 https://github.com/RT-Thread/rt-thread/blob/master/bsp/stm32/libraries/HAL_Drivers/drv_common.c#L138 应该可以直接复制过去就可以用 建一个drv_common.c

**Review by Rbb666 — APPROVED** [review]

**Review by mysterywolf — CHANGES_REQUESTED** [review]
请使用rt_hw_delay_us代替自己的us延时函数

**Comment by Rbb666** [comment]
搞一个gd32-arm通用的drv_common.c吧

**Review by mysterywolf — APPROVED** [review]
我先改成调用rt_hw_us_delay 先合并了

**Pull request merged**